### PR TITLE
update jsonrpsee to v18.2

### LINKED ISSUE #583: Update `jsonrpsee` dependency to verion 18.2
# Issue
`reth` is currently using `jsonrpsee v18.2` In order to reuse some of their features, we have to update this dependency (now we are at `v16.2`). The update to v18.2 causes compatibility issues that need to be fixed.

### PR DESCRIPTION
**Pull request by bkolad**
# Description
update jsonrpsee to v18.2 & fixes compatibility issues.

## Linked Issues
- Fixes #583 

## Testing
No changes.

## Docs
No changes.

### CODE REVIEW

File: module-system/sov-modules-api/Cargo.toml [hunk 1]
```diff
@@ -12,6 +12,7 @@ readme = "README.md"
 resolver = "2"
 
 [dependencies]
+jsonrpsee = { workspace = true }
```
Comment (neysofu) [inline module-system/sov-modules-api/Cargo.toml]: This should probably be `optional = true` and gated behind `native`.

File: examples/demo-rollup/src/ledger_rpc.rs [hunk 2]
```diff
@@ -19,39 +22,48 @@ use self::query_args::{extract_query_args, QueryArgs};
...
-        let args: QueryArgs<SlotIdentifier> = extract_query_args(params)?;
-        db.get_slots::<B, T>(&args.0, args.1).map_err(|e| e.into())
+        let args: QueryArgs<SlotIdentifier> = extract_query_args(params).unwrap();
+        db.get_slots::<B, T>(&args.0, args.1)
+            .map_err(|e| to_jsonrpsee_error_object(e, LEDGER_RPC_ERROR))
     })?;
 
     rpc.register_method("ledger_getBatches", move |params, db| {
-        let args: QueryArgs<BatchIdentifier> = extract_query_args(params)?;
+        let args: QueryArgs<BatchIdentifier> = extract_query_args(params).unwrap();
```
Comment (neysofu) [inline examples/demo-rollup/src/ledger_rpc.rs]: Under what situations can these `.unwrap()` panic? And does the whole server crash if they do?

File: module-system/sov-modules-api/Cargo.toml (hunk 1 above)
Comment (bkolad) [reply module-system/sov-modules-api/Cargo.toml]: Good catch, fixed

File: examples/demo-rollup/src/ledger_rpc.rs (hunk 2 above)
Comment (bkolad) [reply examples/demo-rollup/src/ledger_rpc.rs]: Fixed

**Review by neysofu — APPROVED** [review]

**Pull request merged**